File: app/src/main/java/jp/hazuki/yuzubrowser/browser/openable/OpenUrlList.java
package jp.hazuki.yuzubrowser.browser.openable;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

import jp.hazuki.yuzubrowser.webkit.WebBrowser;

public class OpenUrlList implements BrowserOpenable {
    private final List<String> urllist;
    private final int target;

    public OpenUrlList(List<String> url, int target) {
        this.urllist = url;
        this.target = target;
    }

    public OpenUrlList(Parcel source) {
        source.readStringList(this.urllist = new ArrayList<>());
        this.target = source.readInt();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeStringList(urllist);
        dest.writeInt(target);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Parcelable.Creator<OpenUrlList> CREATOR = new Parcelable.Creator<OpenUrlList>() {
        @Override
        public OpenUrlList createFromParcel(Parcel source) {
            return new OpenUrlList(source);
        }

        @Override
        public OpenUrlList[] newArray(int size) {
            return new OpenUrlList[size];
        }
    };

    @Override
    public void open(WebBrowser browser) {
        for (String url : urllist)
            browser.loadUrl(url, target);
    }
}

File: app/src/main/java/jp/hazuki/yuzubrowser/browser/openable/OpenUrlList.kt
package jp.hazuki.yuzubrowser.browser.openable

import android.os.Parcel
import android.os.Parcelable
import jp.hazuki.yuzubrowser.browser.BrowserController
import java.util.*

class OpenUrlList : BrowserOpenable {
    private val urlList: List<String>
    private val target: Int

    constructor(url: List<String>, target: Int) {
        this.urlList = url
        this.target = target
    }

    constructor(source: Parcel) {
        urlList = ArrayList()
        source.readStringList(urlList)
        this.target = source.readInt()
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeStringList(urlList)
        dest.writeInt(target)
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun open(controller: BrowserController) {
        for (url in urlList)
            controller.loadUrl(url, target)
    }

    companion object {

        @JvmField
        val CREATOR: Parcelable.Creator<OpenUrlList> = object : Parcelable.Creator<OpenUrlList> {
            override fun createFromParcel(source: Parcel): OpenUrlList {
                return OpenUrlList(source)
            }

            override fun newArray(size: Int): Array<OpenUrlList?> {
                return arrayOfNulls(size)
            }
        }
    }
}
